utilities/unittests/test_pytest_utils.py

```
@@ -25,9 +26,110 @@
+
+class TestValidateCpuArchParams:
+    """Test cases for validate_cpu_arch_params function"""
+
+    @patch("utilities.pytest_utils.get_cluster_architecture", return_value={"amd64"})
+    def test_homogeneous_cluster_no_option_ok(self, mock_get_cluster_arch):
+        """Test homogeneous cluster with no --cpu-arch option does not raise"""
+        validate_cpu_arch_params(cpu_arch_option="")
+        mock_get_cluster_arch.assert_called_once()
+
+    @patch("utilities.pytest_utils.get_cluster_architecture", return_value={"amd64"})
+    def test_homogeneous_cluster_with_option_raises(self, mock_get_cluster_arch):
```

Ruff (0.14.14)

[warning] Unused method argument: `mock_get_cluster_arch`

(ARG002)